### PR TITLE
support random debug port by assigning port = 0

### PR DESCRIPTION
**Pull request by xujuntwt95329**
If we don't want to specify a debug port, then we can use `-g=127.0.0.1:0` to get a random port

### CODE REVIEW

File: core/iwasm/libraries/debug-engine/gdbserver.c [hunk 1]
```diff
@@ -98,6 +98,14 @@ wasm_launch_gdbserver(char *host, int port)
         goto fail;
     }
 
+    socklen = sizeof(addr);
+    if (getsockname(listen_fd, (void *)&addr, &socklen) == -1) {
+        LOG_ERROR("%s", strerror(errno));
+        goto fail;
+    }
+
+    LOG_VERBOSE("Listening on %s:%d\n", host, ntohs(addr.sin_port));
```
Comment (wenyongh) [inline core/iwasm/libraries/debug-engine/gdbserver.c:107]: Suggest to use LOG_WARNING() instead so user can always see the port number without adding "-v=5" to iwasm, or he won't be able to know the port number without "-v=5".

Comment (xujuntwt95329) [reply core/iwasm/libraries/debug-engine/gdbserver.c]: Done

File: core/iwasm/libraries/debug-engine/gdbserver.c [hunk 2]
```diff
@@ -98,6 +98,14 @@ wasm_launch_gdbserver(char *host, int port)
         goto fail;
     }
 
+    socklen = sizeof(addr);
+    if (getsockname(listen_fd, (void *)&addr, &socklen) == -1) {
+        LOG_ERROR("%s", strerror(errno));
+        goto fail;
+    }
+
+    LOG_WARNING("Listening on %s:%d\n", host, ntohs(addr.sin_port));
```
Comment (wenyongh) [inline core/iwasm/libraries/debug-engine/gdbserver.c:107]: And shall we update the iwasm help and source debug document, to mention that when port is 0, a random port will be used?

Comment (xujuntwt95329) [reply core/iwasm/libraries/debug-engine/gdbserver.c:107]: Done

**Pull request merged**